pkg/controller/obtenantbackup_controller.go

```diff
@@ -41,24 +51,203 @@ type OBTenantBackupReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the OBTenantBackup object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
+
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
 func (r *OBTenantBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	crJob := &v1alpha1.OBTenantBackup{}
+	if err := r.Get(ctx, req.NamespacedName, crJob); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	switch crJob.Spec.Type {
+	case v1alpha1.BackupJobTypeFull:
+		fallthrough
```

Comment:
why use fallthrough, the type should be one of these 4

Reply:
It shows that the code treat Full type backup job and Incremental type backup job the same. When control flow enters BackupJobTypeFull branch, it will just enter the following branch then.